selfdrive/car/isotp_parallel_query.py

```diff
@@ -16,17 +16,20 @@ def __init__(self, sendcan, logcan, bus, addrs, request, response, response_offs
     self.request = request
     self.response = response
     self.debug = debug
-    self.functional_addr = functional_addr
     self.response_pending_timeout = response_pending_timeout
 
-    self.real_addrs = []
-    for a in addrs:
-      if isinstance(a, tuple):
-        self.real_addrs.append(a)
-      else:
-        self.real_addrs.append((a, None))
+    if functional_addr:
+      assert all([a in FUNCTIONAL_ADDRS for a in addrs]), "Non-functional addresses in addrs"
+      real_addrs = []
+      if 0x7DF in addrs:
+        real_addrs.extend([(0x7e0 + i, None) for i in range(8)])
+      if 0x18DB33F1 in addrs:
+        # TODO: is this correct?
+        real_addrs.extend([(0x18da00f1 + (i << 8), None) for i in range(256)])
```

Comment:
@gregjhogan do you have a preference how we handle processing all responses when functional querying? I personally like making the rest of the library agnostic to whether we've functional or physical queried, so we can re-use the same logic for physical querying.

Also, are the ranges correct? It might be nice to put them in uds.py, either as a global variable, or make `FUNCTIONAL_ADDRS` a dict

Reply:
Looks good to me.  I think you correctly identified the standard addresses, and we can just add more if we find any additional non-standard addrs we need in the future.